### PR TITLE
fixes error with text encodinf of Manifest

### PR DESCRIPTION
**Pull request by lordofthejars**
@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Would be great if you clarify what was the problem

**Comment by lordofthejars** [comment]
@oleg-nenashev the readManifest logic reads the content of MANIFEST.MF and passes to `Manifest` class in form of `StringInputStream`. The problem is that depending on the encodings of where you are running the logic, you get an invalid header exception. To fix this you need to pass the string to byte array using UTF-8 encoding and then pass the result to `Manifest`

**Comment by oleg-nenashev** [comment]
OK, :bee:

File: src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/mf/ReadManifestStepExecution.java [hunk 1]
```diff
@@ -74,7 +75,7 @@ protected SimpleManifest run() throws Exception {
     }
 
     private SimpleManifest parseText(String text) throws IOException {
-        Manifest manifest = new Manifest(new StringInputStream(text));
+        Manifest manifest = new Manifest(new ByteArrayInputStream(text.getBytes("UTF-8")));
```
Comment (stephenc) [inline src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/mf/ReadManifestStepExecution.java]: Manifest files should always be encoded as UTF-8, so this is good

**Comment by stephenc** [comment]
:bee:

**Pull request merged**